### PR TITLE
`toBeFloat()`

### PR DESCRIPTION
**Pull request by nouraellm**
This commit introduced an new function ToBeFloat() that checks wheter the value is float or not

### CODE REVIEW

File: src/Expectation.php [hunk 1]
```diff
@@ -90,4 +90,41 @@ public function toBePowerOf(int $number): PestExpectation
...
+                $hasExponent = true;
+                $hasPrecision = true;
+
+                continue;
+            }
+
+            return expect(false)->toBeTrue();
+        }
+
+        return expect($hasPrecision)->toBeTrue();
```
Comment (faissaloux) [inline src/Expectation.php:128]:
```suggestion
        return expect(is_float($this->value))->toBeTrue();
```

File: tests/toBeFloat.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+use PHPUnit\Framework\ExpectationFailedException;
+
+it('passes', function ($number): void {
+    expect($number)->toBeFloat();
+})->with([10.0, 28.5, 3.14, -7.1, 0.99]);
```
Comment (faissaloux) [inline tests/toBeFloat.php:7]: Can you add some scientific notations here please?

**Review by faissaloux — CHANGES_REQUESTED** [review]
Thanks @nouraellm for your work on this!

File: src/Expectation.php (hunk 1 above)
Comment (faissaloux) [reply src/Expectation.php:128]: Since we have php's `is_float()`, we can use it here.

Comment (nouraellm) [reply src/Expectation.php:128]: Well ``is_float`` was my initial guess, but I thought you didn't want to use built-in functions?

Comment (faissaloux) [reply src/Expectation.php:128]: No, I don't mind. It's always better to use built-in functions.

**Review by faissaloux — APPROVED** [review]
Thank you!

**Pull request merged**